python/meterpreter/meterpreter.py

```diff
@@ -1504,7 +1518,7 @@ def _core_native_arch(self, request, response):
     def _core_patch_url(self, request, response):
         if not isinstance(self.transport, HttpTransport):
             return ERROR_FAILURE, response
-        new_uri_path = packet_get_tlv(request, TLV_TYPE_TRANS_URL)['value']
+        new_uri_path = packet_get_tlv(request, TLV_TYPE_C2_URL)['value']
```

Comment:
On framework side we are sending:
```ruby
        pkt = Rex::Post::Meterpreter::Packet.new(Rex::Post::Meterpreter::PACKET_TYPE_RESPONSE, Rex::Post::Meterpreter::COMMAND_ID_CORE_PATCH_UUID)
        pkt.add_tlv(Rex::Post::Meterpreter::TLV_TYPE_C2_UUID, conn_id.gsub(/\//, ''))
        resp.body = pkt.to_r
        resp.body = self.c2_profile.wrap_outbound_get(resp.body) if self.c2_profile
```

so this should be changed to this.
```suggestion
        new_uri_path = packet_get_tlv(request, TLV_TYPE_C2_UUID)['value']
```

this is the error i see on the python:

```
DEBUG:root:[-] no request ID found
DEBUG:root:[*] running method core_patch_url
DEBUG:root:[-] method core_patch_url resulted in an error
DEBUG:root:Traceback (most recent call last):
  File "<string>", line 1731, in create_response
  File "<string>", line 1521, in _core_patch_url
KeyError: 'value'
```

Also consider if you do this fix you will see we get an error in the uri set, probably we need to wrap everything with URLencode because we have the `-` character. I already had these issues while working on the python porting., 
also the function should be renamed `_core_patch_uuid` as new standard.

Reply:
Good catch. This should be fixed now in the latest commit. Thanks for testing!